src/vue/common/EmailGetter.spec.js

```diff
@@ -1,116 +1,105 @@
+    let mockHelper
+    let balancesResource
+    let usersResource
 
-  let wrapper
+    let wrapper
 
-  beforeEach(() => {
-    mockHelper = new MockHelper()
-    balancesResource = mockHelper.getHorizonResourcePrototype('balances')
-    usersResource = mockHelper.getApiResourcePrototype('users')
+    beforeEach(() => {
+      mockHelper = new MockHelper()
+      balancesResource = mockHelper.getHorizonResourcePrototype('balances')
+      usersResource = mockHelper.getApiResourcePrototype('users')
 
-    sinon.stub(EmailGetter, 'created').resolves()
-    wrapper = shallowMount(EmailGetter, {
-      localVue
+      sinon.stub(EmailGetter, 'created').resolves()
+      wrapper = shallowMount(EmailGetter, {
+        localVue
+      })
     })
-  })
-
-  afterEach(() => {
-    sinon.restore()
-  })
-
-  it('loadResult() method is called inside created hook only when account ID or balance ID is present', async () => {
-    EmailGetter.created.restore()
-    sinon.stub(EmailGetter.methods, 'isAccountIdOrBalanceIdPresent')
-      .returns(true)
-    const spy = sinon.stub(EmailGetter.methods, 'loadResult')
 
-    await shallowMount(EmailGetter, { localVue })
-
-    expect(spy.calledOnce).to.be.true
-  })
-
-  it('showNoAccountOrBalanceIdFallback() method is called inside created hook if neither account ID nor balance ID is present', async () => {
-    EmailGetter.created.restore()
-    sinon.stub(EmailGetter.methods, 'isAccountIdOrBalanceIdPresent')
-      .returns(false)
-    const spy =
-      sinon.stub(EmailGetter.methods, 'showNoAccountOrBalanceIdFallback')
+    afterEach(() => {
+      sinon.restore()
+    })
 
-    await shallowMount(EmailGetter, { localVue })
+    it('loadResult() is called inside created hook only when account ID or balance ID is present', async () => {
+      EmailGetter.created.restore()
+      const spy = sinon.stub(EmailGetter.methods, 'loadResult')
 
-    expect(spy.calledOnce).to.be.true
-  })
+      await shallowMount(EmailGetter, {
+        localVue,
+        propsData: {
+          accountId: ACCOUNT_ID
+        }
+      })
 
-  it('isAccountIdOrBalanceIdPresent() returns true if either account ID or balance ID is present in props', () => {
-    wrapper.setProps({
-      balanceId: BALANCE_ID
+      expect(spy.calledOnce).to.be.true
     })
 
-    expect(wrapper.vm.isAccountIdOrBalanceIdPresent()).to.be.true
-  })
+    it('loadResult() sets the result property to either account ID or balance ID if an error was thrown', async () => {
+      wrapper.setProps({
+        accountId: ACCOUNT_ID
+      })
+      sinon.stub(wrapper.vm, 'getAccountId').resolves(ACCOUNT_ID)
+      sinon.stub(usersResource, 'get').throws()
+      sinon.stub(console, 'error')
+
+      await wrapper.vm.loadResult()
 
-  it('loadResult() sets the result property to either account ID or balance ID if an error was thrown', async () => {
-    wrapper.setProps({
-      accountId: ACCOUNT_ID
+      expect(wrapper.vm.result).to.equal(ACCOUNT_ID)
```

Comment:
by reading the test I assume that the value resolved by `getAccountId` is assigned to `result` eventually, but actually it isn’t. I’d recommend throwing an error on `getAccountId` call because `catch` does’t care on each step the error is thrown. Find a way to check assignment of account id and after that of balance id less confusing.

Reply:
Resolved, updated